src/components/calendar/bl-calendar.ts

```diff
@@ -87,24 +60,45 @@ export default class BlCalendar extends LitElement {
   @state()
   private _calendarDays: CalendarDay[] = [];
 
-  private _defaultValue: Date | Date[];
-
-  /**
-   * Defines the default selected date value for the calendar
-   */
-  @property({ type: Array, attribute: "default-value", reflect: true })
-  get defaultValue(): Date | Date[] {
-    return this._defaultValue;
-  }
-  set defaultValue(defaultValue) {
-    if (this.type === CALENDAR_TYPES.SINGLE && Array.isArray(defaultValue)) {
-      console.warn("Invalid prop value for defaultValue");
-    } else if (this.defaultValue) {
-      if (Array.isArray(this.defaultValue)) {
-        this._selectedDates = { ...this.defaultValue };
-      } else this._selectedDates = [this.defaultValue];
+  @property({ attribute: "default-value", reflect: true })
+  set defaultValue(defaultValue: Date | Date[]) {
+    if (!this._defaultValue) {
+      if (this.type === CALENDAR_TYPES.SINGLE && Array.isArray(defaultValue)) {
+        console.warn("Invalid prop value.Default value should be Date");
+      } else if (
+        (this.type === CALENDAR_TYPES.MULTIPLE || this.type === CALENDAR_TYPES.RANGE) &&
+        !Array.isArray(defaultValue)
+      ) {
+        console.warn("Invalid prop value.Default value should be Date array");
+      } else if (
+        this.type === CALENDAR_TYPES.RANGE &&
+        Array.isArray(defaultValue) &&
+        (defaultValue.length < 0 || defaultValue.length > 2)
+      ) {
+        console.warn("Invalid prop value.Default value should be two Date items");
```

Comment:
"'defaultValue' must be an array of two Date objects when the date selection mode is set to range." [optional]